### PR TITLE
ENHANCEMENT Allow menu to be extendable from DNProject

### PR DESCRIPTION
**Pull request by halkyon**
Allows an extension class to add their own menu items and to remove
existing ones if necessary.

### CODE REVIEW

File: code/model/DNProject.php [hunk 1]
```diff
@@ -148,6 +148,33 @@ public function HasDiskQuota() {
...
+	public function Menu() {
+		$list = new ArrayList();
+
+		$list->push(new ArrayData(array(
+			'Link' => sprintf('naut/project/%s', $this->Name),
+			'Title' => 'Deploy',
+			'IsActive' => Controller::curr()->getAction() == 'project'
+		)));
+
+		if(Controller::curr()->FlagSnapshotsEnabled()) {
```
Comment (tractorcow) [inline code/model/DNProject.php]: What if the current controller isn't DNRoot (e.g. in the case of a nested controller?). How about `singleton('DNRoot')` or just making it a static method?

Comment (halkyon) [inline code/model/DNProject.php]: Static seems fine, and it's not used anywhere else from what I can tell :) have rebased the change.

**Comment by stojg** [comment]
I think it looks good enough to support adding more menu items, which we will have to do unless we want to bundle everything into deploynaut base module, which I don't like at all.

In the end I think we will have to do some UI refactoring around the navigation and presentation. Deploynauts look and feel could need some :heart:

**Comment by tractorcow** [comment]
Nice job. :)

**Pull request merged**